src/main/java/seedu/module/logic/commands/AddDeadlineCommand.java

```diff
@@ -0,0 +1,83 @@
+package seedu.module.logic.commands;
+
+import static seedu.module.logic.parser.CliSyntax.PREFIX_DEADLINE;
+
+import java.util.List;
+
+import seedu.module.commons.core.Messages;
+
+import seedu.module.commons.core.index.Index;
+import seedu.module.logic.commands.exceptions.CommandException;
+import seedu.module.model.Model;
+import seedu.module.model.module.ArchivedModule;
+import seedu.module.model.module.Deadline;
+import seedu.module.model.module.TrackedModule;
+
+
+/**
+ * Adds deadline to be module.
+ */
+public class AddDeadlineCommand extends Command {
+
+    public static final String COMMAND_WORD = "deadline";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a deadline to a specific module. "
+            + "Parameters: "
+            + "INDEX (must be a positive integer) "
+            + PREFIX_DEADLINE + "DESCRIPTION\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + PREFIX_DEADLINE + " quiz submission /by 2/2/2019 2359";
+
+    public static final String MESSAGE_ADD_DEADLINE_SUCCESS = "Added deadline to Module: %1$s";
+    public static final String MESSAGE_DELETE_DEADLINE_SUCCESS = "Removed remark from Person: %1$s";
+
+    private final Index index;
+    private final Deadline deadline;
+
+    public AddDeadlineCommand(Index index, Deadline deadline) {
+        this.index = index;
+        this.deadline = deadline;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        List<TrackedModule> lastShownList = model.getFilteredModuleList();
+
+        if (index.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_MODULE_DISPLAYED_INDEX);
+        }
+
+        TrackedModule moduleToEdit = lastShownList.get(index.getZeroBased());
+
+        ArchivedModule archivedModule = new ArchivedModule(moduleToEdit.getModuleCode(),
+                moduleToEdit.getTitle(), moduleToEdit.getDescription());
```

Comment:
Is there a need to clone `ArchivedModule` here? The archived module is read-only so you only need to copy the reference.